drivers/hmis_csv_twenty_twenty_six/app/models/hmis_csv_twenty_twenty_six/importer/custom/column_mapper.rb

```diff
@@ -301,238 +247,77 @@ def self.apply_mappings(source_record, mapped_attributes, column_configs, lookup
     end
 
-    # Checks if any column configurations require record lookups
-    #
-    # @param column_configs [Array<Hash>] Array of column configurations from YAML
-    # @return [Boolean] True if any columns use record_lookup mapping type
-    def self.record_lookups?(column_configs)
-      column_configs.any? { |col| col.dig('warehouse_column_mapping', 'type') == 'record_lookup' }
-    end
-
-    # Applies record lookup mapping with optional caching for batch processing
-    #
-    # @param source_record [Object] Source record with data to map
-    # @param mapped_attributes [Hash] Hash to store mapped attributes
-    # @param mapping_config [Hash] Configuration for the record lookup
-    # @param lookup_cache [Hash, nil] Optional cache for record lookups
-    # @return [void]
-    # @private
-    private_class_method def self.apply_record_lookup_mapping(source_record, mapped_attributes, mapping_config, lookup_cache)
-      class_column = mapping_config['class_column']
-      target_column = mapping_config['target_column']
-      lookup_field_mappings = mapping_config['lookup_field_mappings']
-      source_column = mapping_config['source_column'] || 'RecordID'
-
-      # Get the source values
-      source_value = source_record[source_column]
-      class_name = mapped_attributes[class_column]
-      data_source_id = source_record.data_source_id
-
-      if source_value.blank? || class_name.blank?
-        mapped_attributes[target_column] = nil
-        return
-      end
-
-      lookup_field = lookup_field_mappings[class_name]
-      if lookup_field.blank?
-        Rails.logger.warn "No lookup field mapping found for class: #{class_name}"
-        mapped_attributes[target_column] = nil
-        return
-      end
-
-      if lookup_cache
-        # Use cache for batch processing
-        cache_key = [class_name, data_source_id, lookup_field, source_value]
-        database_id = lookup_cache[cache_key]
-
-        if database_id.nil? && !lookup_cache.key?(cache_key)
-          # Cache miss - this shouldn't happen in proper batch processing
-          Rails.logger.warn "Cache miss during record lookup: #{cache_key}"
-          database_id = perform_individual_lookup(class_name, lookup_field, source_value, data_source_id)
-          lookup_cache[cache_key] = database_id
-        end
-      else
-        # Individual lookup (less efficient)
-        database_id = perform_individual_lookup(class_name, lookup_field, source_value, data_source_id)
+    # Casts a value to the specified type.
+    # @param value [Object] The value to cast.
+    # @param type [String] The target type (e.g., 'integer', 'string').
+    # @return [Object] The casted value.
+    private def cast_value(value, type)
+      # Preserve boolean false, since `false.blank?` is true.
+      return value if value == false
+      return nil if value.blank?
+
+      case type
+      when 'integer'
+        value.to_i
+      else # Default to string or no-op
+        value
       end
```

Comment:
maybe worth being a little more defensive?

```suggestion
      # Default to 'string' if type is not specified in the YAML config.
      type_to_cast = type || 'string'

      case type_to_cast
      when 'integer'
        value.to_i
      when 'date'
        value.to_date
      when 'datetime'
        Time.zone.parse(value)
      when 'string'
        value
      else
        raise ArgumentError, "Unknown column type '#{type}' specified in custom file configuration."
      end
```